detect_os

```diff
@@ -119,6 +121,11 @@ parse_uname()
             echo $identifier has no uname_version_pattern, skipping
             continue
         fi
+
+	if [ -z pattern ]; then
```

Comment:
Shouldn't this start with a `$`?

So this:
```suggestion
	if [ -z "$pattern" ]; then
```

I think the uname pattern has the same problem

Reply:
I thought it looked wrong as well but as you said the uname pattern is wrong and I was just copying what was there assuming it was correct.  Want me to fix both at the same time or leave the other so as to not mix issues in the same push?

Reply:
Go ahead and fix both, no reason to delay this small of a fix/bug